pulsar-testclient/src/main/java/org/apache/pulsar/testclient/PerformanceProducer.java

```diff
@@ -497,7 +500,7 @@ private static void runProducer(Arguments arguments,
                         }
                     }).exceptionally(ex -> {
                         log.warn("Write error on message", ex);
-                        System.exit(-1);
+                        messagesFailed.increment();
```

Comment:
Can we also retain a mode to fail on error? I found it very useful to leave the perf producer running as a system test for hours and check performance. In these cases it's desirable to assert no error was observed in the entire run.

Reply:
> Can we also retain a mode to fail on error?

we can add a flag to continue on failure.. default we can keep it to exit from the process.

Reply:
I'm fine with changing the default

Reply:
added a flag to exit on failure. default is disable and it continues.